### PR TITLE
Add user facing option to place blank lines between using/import groups.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/20988

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @kuhlenh

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
tagging @dpoeschl Roslyn.VisualStudio.IntegrationTests.VisualBasic.BasicLineCommit.CommitOnSave is failing.

**Comment by dpoeschl** [comment]
retest windows_debug_vs-integration_prtest please

**Comment by CyrusNajmabadi** [comment]
Tagging @dpoeschl https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_unit32_prtest/8588/

File: src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesOrganizer.cs [hunk 1]
```diff
@@ -2,21 +2,95 @@
...
+            }
+
+            if (directive1IsUsingStatic && directive2IsUsingStatic)
+            {
+                return false;
+            }
+
+            if (directive1IsNamespace && directive2IsNamespace)
+            {
+                // Both normal usings.  Place them in groups if they're first namespace
```
Comment (dpoeschl) [inline src/Workspaces/CSharp/Portable/Utilities/UsingsAndExternAliasesOrganizer.cs]: they're --> their

File: src/Workspaces/Core/Portable/Editing/GenerationOptions.cs [hunk 2]
```diff
@@ -6,9 +6,16 @@ namespace Microsoft.CodeAnalysis.Editing
...
+        public static readonly PerLanguageOption<bool> PlaceSystemNamespaceFirst = new PerLanguageOption<bool>(nameof(GenerationOptions), 
+            nameof(PlaceSystemNamespaceFirst), defaultValue: true,
             storageLocations: new OptionStorageLocation[] {
                 EditorConfigStorageLocation.ForBoolOption("dotnet_sort_system_directives_first"),
                 new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.PlaceSystemNamespaceFirst")});
+
+        public static readonly PerLanguageOption<bool> SeparateImportDirectiveGroups = new PerLanguageOption<bool>(
+            nameof(GenerationOptions), nameof(SeparateImportDirectiveGroups), defaultValue: false,
+            storageLocations: new OptionStorageLocation[] {
+                EditorConfigStorageLocation.ForBoolOption("dotnet_seperate_import_directive_groups"),
```
Comment (dpoeschl) [inline src/Workspaces/Core/Portable/Editing/GenerationOptions.cs]: seperate -> separate (this one actually matters)

**Pull request merged**